evm/auxiliary/user_account.py

```diff
@@ -0,0 +1,181 @@
+SIGNATURE_S_SLICE = slice(2 * 32, 3 * 32)
+SIGNATURE_SLICE = slice(SIGNATURE_V_SLICE.start, SIGNATURE_S_SLICE.stop)
+NONCE_SLICE = slice(3 * 32, 4 * 32)
+GASPRICE_SLICE = slice(4 * 32, 5 * 32)
+VALUE_SLICE = slice(5 * 32, 6 * 32)
+RECEIVER_SLICE = slice(6 * 32, 7 * 32)
+DATA_SLICE = slice(7 * 32, None)  # open ended
+
+EMPTY_DATA = b'\x00' * RECEIVER_SLICE.stop
+
+
+class StandardTransaction(ShardingTransaction):
+    """A transaction calling the standard user account contract.
+
+    The contract expects transaction data to be formatted as:
+        # 0-95: sig
+        # 96-127: nonce
+        # 128-159: gasprice
+        # 160-191: value
+        # 192-223: to
+        # 224+: data
+    """
+
+    def __init__(
+        self,
+        chain_id,
+        shard_id,
+        target,
+        start_gas,
+        access_list,
+        receiver,
+        value,
+        nonce,
+        gas_price,
+        msg_data
+    ):
+        super().__init__(
+            chain_id=chain_id,
+            shard_id=shard_id,
+            target=target,
+            data=EMPTY_DATA,
+            start_gas=start_gas,
+            access_list=access_list,
+            code=b'',
+        )
+        self.receiver = receiver
+        self.value = value
+        self.nonce = nonce
+        self.gas_price = gas_price
+        self.msg_data = msg_data
+
+    #
+    # Properties for fields encoded in data
+    #
+
+    @property
+    def signature(self):
+        v = big_endian_to_int(self.data[SIGNATURE_V_SLICE])
+        r = big_endian_to_int(self.data[SIGNATURE_R_SLICE])
+        s = big_endian_to_int(self.data[SIGNATURE_S_SLICE])
+        return (v, r, s)
+
+    @signature.setter
+    def signature(self, vrs):
+        for i in vrs:
+            validate_uint256(i)
+
+        b = b''.join([
+            pad32(int_to_big_endian(i)) for i in vrs
+        ])
+
+        self.data = self.data[:SIGNATURE_SLICE.start] + b + self.data[SIGNATURE_SLICE.stop:]
+
+    @property
+    def nonce(self):
+        return big_endian_to_int(self.data[NONCE_SLICE])
+
+    @nonce.setter
+    def nonce(self, value):
+        validate_uint256(value)
+        b = pad32(int_to_big_endian(value))
+        self.data = self.data[:NONCE_SLICE.start] + b + self.data[NONCE_SLICE.stop:]
+
+    @property
+    def gas_price(self):
+        return big_endian_to_int(self.data[GASPRICE_SLICE])
+
+    @gas_price.setter
+    def gas_price(self, value):
+        validate_uint256(value)
+        b = pad32(int_to_big_endian(value))
+        self.data = self.data[:GASPRICE_SLICE.start] + b + self.data[GASPRICE_SLICE.stop:]
+
+    @property
+    def value(self):
+        return big_endian_to_int(self.data[VALUE_SLICE])
+
+    @value.setter
+    def value(self, value_):
+        validate_uint256(value_)
+        b = pad32(int_to_big_endian(value_))
+        self.data = self.data[:VALUE_SLICE.start] + b + self.data[VALUE_SLICE.stop:]
+
+    @property
+    def receiver(self):
```

Comment:
This fragments the terminology that we use for the *destination* of a transaction.  Elsewhere we're using `to`.  Any thoughts on whether it'd be better to conform to this naming convention?

Reply:
Not sure... We used to have `to` and `sender`. Now we have `target`, `sender` and `receiver`.

- `to` and `target` refer to more or less the same thing I guess (the first account that is called)
- `sender` changes meaning: it used to be the one who signed the transaction, now it's the "owner" of the `target` account (and just happens to put a signature in the `data` field)
- `receiver` is the second hop (the account that's called by the `target` account)

So from a technical standpoint, I don't see a lot of overlap between original and new terms. I could imagine renaming `target` to `to` for consistency (although I'd prefer `target` in a vacuum). Renaming `receiver` to `to` could be misleading on the protocol level, but would make sense from a high level point of view in most cases. I don't see a good alternative to `sender`.

Reply:
Good explanation.  I think you are right to leave this as-is and not change the names from `target/sender/receiver`

Reply:
Does the name `destination` better capture the concept that `receiver` currently represents?